### PR TITLE
Add function to find the nearest point and interpolate

### PR DESCRIPTION
**Pull request by ralucaportase**

### CODE REVIEW

File: tests/jquery.flot.Test.js [hunk 1]
```diff
@@ -462,6 +462,44 @@ describe('flot', function() {
         });
     });
 
+    describe('findNearbyInterpolationPoint', function() {
+        var placeholder, plot, sampledata = [[0, 1], [1, 1.1], [2, 1.2]];
```
Comment (cristiingineru) [inline tests/jquery.flot.Test.js]: Could you please add 2 more tests: one for an empty sampledata and another one with a single point.

File: jquery.flot.js [hunk 2]
```diff
@@ -2583,7 +2583,11 @@ Licensed under the MIT license.
 
                 if ((p1x === undefined) || (p2x === undefined) ||
                     (p1y === undefined) || (p2y === undefined)) {
-                    continue;
+                    if (points.length === 2) {
```
Comment (cristiingineru) [inline jquery.flot.js]: points.length === ps ?

File: tests/jquery.flot.Test.js [hunk 3]
```diff
@@ -498,6 +498,24 @@ describe('flot', function() {
...
+            expect(item).toEqual(null);
+        });
+
+        it('for a dataserie with a single plot should return that point', function() {
+            plot = $.plot(placeholder, [[[1, 2]]], {});
+            var item = plot.findNearbyInterpolationPoint(0, 0, function() {
+                return true;
+            });
+
+            expect(item.datapoint).toEqual([1, 2]);
```
Comment (cristiingineru) [inline tests/jquery.flot.Test.js]: Are you sure about this? Wouldn't the cursor snap on both x and y in this case?
